File: solutions/python/scrabble-score/1/scrabble_score.py
def score(word):
    word=word.lower()
    score=0
    for ch in word:
        print(ch)
        match ch:
            case 'a' | 'e' | 'i' | 'o' | 'u' | 'l' | 'n' | 'r' | 's' | 't' :
                score=score+1
            case 'd' | 'g' :
                score=score+2
            case 'b' | 'c' | 'm' | 'p' :
                score=score+3
            case 'f' | 'h' | 'v' | 'w' | 'y' :
                score=score+4
            case 'k' :
                score=score+5
            case 'j' | 'x' :
                score=score+8
            case 'q' | 'z' :
                score=score+10
            case default:
                score=score
   
    return score

File: solutions/c/scrabble-score/1/scrabble_score.c
#include "scrabble_score.h"
#include <ctype.h>

unsigned int score(const char *word){
    unsigned int score=0;
    char points[]="02210313074020029000033739";
    for(int i=0;i<30;i++){
        if ((*word)=='\0'){
            i=31;
        }
        else {
            score=score+points[tolower(*word)-97]-47;
            word++;          
        }
    }
    return score; 
}
